### DISCUSSION TITLE
Can I enlarge the screen content from the centre?

### DISCUSSION THREAD

**Discussion by HubKing** [dosbox-staging/dosbox-staging#2958] [Q&A] [answered]
Is it possible to enlarge the screen content like below so that only the central part of the screen is displayed on the monitor?

![image](https://github.com/dosbox-staging/dosbox-staging/assets/10900489/cf2efb8e-dcf4-4aa7-9281-92e504581043)

**Comment by GranMinigun** [comment]
No, filling the screen is not possible. What's your use-case for that?

**Reply by HubKing** [reply]
Yeah, after posting this, I found the viewport zoom option but it only allowed a value less than 100%. If it can already resize the viewport, why doesn't it support making it bigger? Is it a technical limitation, or an arbitrary disposition thinking that no one would want that?

The use case is that some game's content area is 16:9, DOS resolution is 4:3, I guess, so this game would have showed letter boxes back in its days. Now thay my monitor is 16:9, in full screen mode, this makes a black bar on all four sides.

**Reply by kcgen** [reply]
@HubKing , what is your host display's  resolution?

**Reply by HubKing** [reply]
Well, I have multiple computers. For example, the laptop has 1920 * 1080. The game was a Japanese game, converted from its original PC98 version. PC98 seems to have had the resolution 640 * 400, so probably that is why the DOS port has top/bottom black bars.

**Reply by johnnovak** [reply]
But in that case if say the graphics uses a 640x400 area of the 640x480 screen, and you "zoom it in" so vertically it fills the screen, content on the two sides would be cut off. Like those old DVD releases of widescreen movies that "converted" the movie to 4:3 and cut off a lot of content.

> Now thay my monitor is 16:9, in full screen mode, this makes a black bar on all four sides.

On all four sides? Are you using the latest dev build with CRT shaders enabled? That auto-enables integer scaling.

Anyway, screenshots, logs, and your config please, this is too vague.

**Reply by HubKing** [reply]
Have you seen an old 4:3 YouTube video of a 16:9 movie? If you press the full-screen button on your 16:9 monitor, you will have black bars on all sides, because the top/bottom black bars are part of the YouTube video.

Anyway, here is the game in window mode.
![window](https://github.com/dosbox-staging/dosbox-staging/assets/10900489/65cc30e1-9f68-4020-97b2-5074fb29ff39)

Here is the same game in full-screen mode (you need not be using a dark theme of GitHub to see the black bars):
![fullscreen](https://github.com/dosbox-staging/dosbox-staging/assets/10900489/ca14f44a-ed9a-4bf4-9691-67e51b672b8a)

Actually, you can see it yourself within the web browser at https://playclassic.games/games/role-playing-dos-games-online/play-new-horizons-online/play . Click the full screen button below the page, and press ESC to skip the intro. After the intro, select "New Game" and you will see the same screen as my screen captures above.

**Reply by johnnovak** [reply]
Yeah OK, now I got it. I don't think we want to support this use case, seems like unnecessary complication and frankly, the letterboxing is very minimal. The problem is, even an innocent looking feature complicates things when you need to consider compatibility with other features, then complexity becomes hard to manage...

I'm especially worried how supporting greater than 100% "viewport sizes" would complicate the upcoming border support feature. Plus it would probably complicate the adaptive shader switching mechanism, and possibly the integer scaling features, or at least make them really unintuitive when you go beyond the physical dimensions of your screen or window.

All in all, my suggestion is to live with it. Personally, I don't even want fullscreen in any low resolution game; they look way better if the physical dimension of the image resembles that of a typical 90s CRT monitor (which rarely went above 17"), otherwise the pixels are just too big... The screenshot looks good to me on my 24" screen, but I know this is subjective.

I _could_ be convinved if someone thinks through all the implications and how it could work together with existing features, consider the edge cases, etc... Frankly, I have neither the time nor the inclination to do that now for such a niche feature.

**Reply by kcgen** [reply]
@HubKing - the `windowresolution = ` conf setting will "do as requested".
So, provided your OS permits windows larger than your logical 1080p desktop, then you can over-blow the window to something like:

`windowresolution = 2560x1440`

Then simply click 'n drag to center accordingly; maybe a workable compromise.

**Reply by HubKing** [reply]
If I remember it correctly, under Windows, I could move a window out of the screen boundary (using the "Move" menu of the system menu). I am using Gnome now, and it seems that Gnome does not allow that. I could fit the context area horizontally, but I could not move the window above the upper screen boundary.

**Reply by rderooy** [reply]
Try pressing Alt-F7 and you should be able to move the window over the upper screen boundary. It works here for Gnome 45.

**Reply by HubKing** [reply]
You are right. With the move shortcut key, it seems that Gnome allows the mouse cursor go above the screen area, unlike when the move was initiated by the "Move" menu of the window context menu. I wonder why they act differently. A bug? They should act the same way, in my opinion.

**Comment by johnnovak** [comment, marked as answer]
@HubKing 

You can achieve the zooming now with the new `viewport = relative` setting we introduced in v0.81.0.

More details:

- https://dosbox-staging.github.io/releases/release-notes/0.81.0/#aspect-ratio-and-viewport-enhancements

- https://dosbox-staging.github.io/getting-started/advanced-graphics-options/#custom-viewport-resolution

**Reply by HubKing** [reply]
I have reset the .conf file and tried the following (the aspect ratio was correct in the first place, so I kept it `on`, and used the same relative values to both w/h). The output size/location were what I had wanted, but the solid-filled area shows noticeable fine double-horizontal lines. Is this an inevitable artifact of this type of scaling or can I somehow mitigate these lines?

aspect             = on
integer_scaling    = off
viewport           = relative 120% 120%
monochrome_palette = amber
cga_colors         = default
glshader           = crt-auto

**Reply by johnnovak** [reply]
> The output size/location were what I had wanted, but the solid-filled area shows noticeable fine double-horizontal lines. Is this an inevitable artifact of this type of scaling or can I somehow mitigate these lines?

That's VGA double scanning. You've enlarged the image, so the emulated scanlines are bigger, of course. The exact same thing would happen on a real CRT monitor.

If you don't want scanlines and CRT emulation, use `glshader = sharp`.
